src/core/lib/security/credentials/oauth2/oauth2_credentials.cc

```diff
@@ -252,6 +259,7 @@ void grpc_oauth2_token_fetcher_credentials::on_http_response(
     if (status == GRPC_CREDENTIALS_OK) {
       grpc_credentials_mdelem_array_add(pending_request->md_array,
                                         access_token_md);
+      add_additional_metadata(pending_request->md_array);
```

Comment:
The fact that we need make sure to use it both when token is cached and when it's not cached is slightly subtle.

Is it possible to unconditionally add the "additional metadata" e.g. at the top of `grpc_oauth2_token_fetcher_credentials::get_request_metadata` ?

Reply:
Good point, done. PTAL